drivers/storage/portworx/manifest/manifest.go

```diff
@@ -0,0 +1,134 @@
+package manifest
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"sort"
+
+	version "github.com/hashicorp/go-version"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	// ManifestDir directory where release manifest is stored
+	ManifestDir = "manifests"
+	// ReleasesFilename name of the release manifest file
+	ReleasesFilename = "portworx-releases.yaml"
+)
+
+var (
+	// ErrEmptyReleases when releases list is empty
+	ErrEmptyReleases = errors.New("releases cannot be empty")
+	// ErrReleaseNotFound when given release is not found
+	ErrReleaseNotFound = errors.New("release not found")
+	// ErrInvalidDefaultRelease when the default release is invalid
+	ErrInvalidDefaultRelease = errors.New("invalid default release")
+	loadReleaseManifest      = readManifestFile
+)
+
+// ReleaseManifest defines the Portworx release manifest
+type ReleaseManifest struct {
+	DefaultRelease string             `yaml:"defaultRelease,omitempty"`
+	Releases       map[string]Release `yaml:"releases,omitempty"`
+}
+
+// Release is a single release object with images for different components
+type Release struct {
+	Stork      string `yaml:"stork,omitempty"`
+	Lighthouse string `yaml:"lighthouse,omitempty"`
+}
+
+// NewReleaseManifest returns a release manifest object from the portworx releases file
+func NewReleaseManifest() (*ReleaseManifest, error) {
+	data, err := loadReleaseManifest()
+	if err != nil {
+		return nil, err
+	}
+	manifest := &ReleaseManifest{}
+	err = yaml.Unmarshal([]byte(data), manifest)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(manifest.Releases) == 0 {
+		return nil, ErrEmptyReleases
+	}
+	if len(manifest.DefaultRelease) == 0 {
+		manifest.DefaultRelease = manifest.getLatest()
+	} else if _, exists := manifest.Releases[manifest.DefaultRelease]; !exists {
+		return nil, ErrInvalidDefaultRelease
+	}
+	return manifest, nil
+}
+
+// Get returns a release object for given string version
+func (m *ReleaseManifest) Get(target string) (Release, error) {
+	if release, exists := m.Releases[target]; exists {
+		return release, nil
+	}
+	targetVersion, err := version.NewSemver(target)
+	if err != nil {
+		return Release{}, fmt.Errorf("invalid version %s: %v", target, err)
```

Comment:
You can return `*Release` from the function, so that you can return `nil, err` instead of an empty Release struct